lib/streamlit/elements/arrow_altair.py

```diff
@@ -395,190 +678,575 @@ def _melt_data(
+) -> pd.DataFrame:
+    """Returns a subset of df, selecting only column_names that aren't None."""
+
+    # We can't just call set(col_names) because sets don't have stable ordering,
+    # which means tests that depend on ordering will fail.
+    # Performance-wise, it's not a problem, though, since this function is only ever
+    # used on very small lists.
+    seen = set()
+    keep = []
+
+    for x in column_names:
+        if x is None:
+            continue
+        if x in seen:
+            continue
+        seen.add(x)
+        keep.append(x)
+
+    return df[keep]
+
+
+def _maybe_convert_color_column_in_place(df: pd.DataFrame, color_column: Optional[str]):
+    """If needed, convert color column to a format Vega understands."""
+    if color_column is None or len(df[color_column]) == 0:
+        return
+
+    first_color_datum = df[color_column][0]
+
+    if is_hex_color_like(first_color_datum):
+        # Hex is already CSS-valid.
+        pass
+    elif is_color_tuple_like(first_color_datum):
+        # Tuples need to be converted to CSS-valid.
+        df[color_column] = df[color_column].map(to_css_color)
+    else:
+        # Other kinds of colors columns (i.e. pure numbers or nominal strings) shouldn't
+        # be converted since they are treated by Vega-Lite as sequential or categorical colors.
+        pass
+
+
+def _convert_col_names_to_str_in_place(
+    df: pd.DataFrame,
+    x_column: Optional[str],
+    y_column_list: List[str],
+    color_column: Optional[str],
+) -> Tuple[Optional[str], List[str], Optional[str]]:
+    """Converts column names to strings, since Vega-Lite does not accept ints, etc."""
+    column_names = list(df.columns)  # list() converts RangeIndex, etc, to regular list.
+    str_column_names = [str(c) for c in column_names]
+    df.columns = pd.Index(str_column_names)
+
+    return (
+        None if x_column is None else str(x_column),
+        [str(c) for c in y_column_list],
+        None if color_column is None else str(color_column),
     )
 
-    opacity = None
-    if chart_type == ChartType.AREA and color_column:
-        opacity = {y_column: 0.7}
+
+def _parse_color_column(
+    df: pd.DataFrame, column_or_value: Any
+) -> Tuple[Optional[str], Any]:
+    if isinstance(column_or_value, str) and column_or_value in df.columns:
+        column_name = column_or_value
+        value = None
+    else:
+        column_name = None
+        value = column_or_value
+
+    return column_name, value
+
+
+def _parse_x_column(df: pd.DataFrame, x_from_user: Optional[str]) -> Optional[str]:
+    if x_from_user is None:
+        return None
+
+    elif isinstance(x_from_user, str):
+        if x_from_user not in df.columns:
+            raise StreamlitColumnNotFoundError(df, x_from_user)
+
+        return x_from_user
+
+    else:
+        raise StreamlitAPIException(
+            "x parameter should be a column name (str) or None to use the "
+            f" dataframe's index. Value given: {x_from_user}"
```

Comment:
can we output the type here rather than the value? It seems like we are checking for str type and would be more handy to print out type to me. Maybe both the value and the type is better but at least type imo.